### PR TITLE
Activemq Metrics Receiver

### PR DESCRIPTION
**Pull request by armstrmi**
Add Support for Activemq Metrics.

### CODE REVIEW

**Comment by armstrmi** [comment]
@qingling128 kokoro has passed, just have had to re-merge with base on multiple occasions

**Comment by armstrmi** [comment]
@qingling128 Could you review this when you get the chance? Thank you!

File: docs/activemq.md
Comment (qingling128) [inline docs/activemq.md:25]: It's preferred for the receiver and pipeline IDs to stay the same as the receiver type in the sample configuration.

aka `s/activemq_metrics/activemq`, `s/activemq_pipeline/activemq`

File: docs/activemq.md
Comment (qingling128) [inline docs/activemq.md:18]: Also `wildcard_refresh_interval`?

e.g.

```
| `wildcard_refresh_interval` | `60s` | The interval at which wildcard file paths in `include_paths` are refreshed. Specified as a time interval parsable by [time.ParseDuration](https://pkg.go.dev/time#ParseDuration). Must be a multiple of 1s.|
```

Comment (armstrmi) [reply docs/activemq.md:18]: I believe that would be covered by the logs receiver config, which will be implemented after this PR, but for now, this is simply for the metrics receiver config

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/install
Comment (armstrmi) [reply integration_test/third_party_apps_data/applications/activemq/centos_rhel/install:72]: The variables should already exist in the `env` file and therefore we felt that we should keep it and only modify the things we need

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/install
Comment (armstrmi) [reply integration_test/third_party_apps_data/applications/activemq/centos_rhel/install:8]: Unfortunately for ActiveMQ there is no listed rpm/deb installer, it only points to zip files in ActiveMQ's [install quide](https://activemq.apache.org/version-5-getting-started)

File: apps/activemq.go
Comment (armstrmi) [reply apps/activemq.go:30]: Once that PR is merged I'm sure we could update

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/install
Comment (quentinmit) [reply integration_test/third_party_apps_data/applications/activemq/centos_rhel/install:8]: There's an `activemq` package in all the supported Debian versions: https://packages.debian.org/search?keywords=activemq and Ubuntu versions: https://packages.ubuntu.com/search?keywords=activemq

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/install
Comment (quentinmit) [reply integration_test/third_party_apps_data/applications/activemq/centos_rhel/install:72]: This is a lot of customization; the install scripts should be the absolute bare minimum necessary to make the app run (ideally, just installing the package). If there's something that needs to be changed in a config file, prefer making a one-line change using e.g. `sed` or an `/etc/defaults` file.

**Review by quentinmit — CHANGES_REQUESTED** [review]

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/install
Comment (qingling128) [reply integration_test/third_party_apps_data/applications/activemq/centos_rhel/install:72]: +1 that `the install scripts should be the absolute bare minimum necessary` to reduce the scope of things we need to maintain in the long term.

File: apps/activemq.go
Comment (qingling128) [reply apps/activemq.go:30]: Yay, #370 has been merged

File: docs/activemq.md
Comment (qingling128) [reply docs/activemq.md:18]: Oops. My mistake. Ignore this.

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/install
Comment (armstrmi) [reply integration_test/third_party_apps_data/applications/activemq/centos_rhel/install:8]: @quentinmit @qingling128 new install scripts applied, should be up to date

**Comment by armstrmi** [comment]
Screen Shot 2022-02-07 at 4 30 02 PM
failing integration test is coming from tomcat, no changes to any technologies has been made, seems to be a kokoro test issue @quentinmit @qingling128

File: apps/activemq.go
Comment (qingling128) [inline apps/activemq.go:34]: Can we use `MetricsReceiverSharedCollectJVM` here? The recent Elasticsearch PR added some common utils: https://github.com/GoogleCloudPlatform/ops-agent/pull/349/files

File: docs/activemq.md
Comment (qingling128) [inline docs/activemq.md:28]: Remove the `collection_interval` here so that we can provide the bare minimum config for users in the sample config?

File: integration_test/third_party_apps_data/agent/ops-agent/linux/enable_activemq
Comment (qingling128) [inline integration_test/third_party_apps_data/agent/ops-agent/linux/enable_activemq:24]: Remove the `endpoint` and `collection_interval` configurations here so we can test the bare minimum, which is referenced by the sample config? See more context in https://github.com/GoogleCloudPlatform/ops-agent/pull/384#issue-1123802764

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/install
Comment (qingling128) [reply integration_test/third_party_apps_data/applications/activemq/centos_rhel/install:8]: Thanks!

**Review by qingling128 — APPROVED** [review]
LGTM with some minor comments.

**Comment by armstrmi** [comment]
Screen Shot 2022-02-08 at 1 42 50 PM
Reason for cla failure @qingling128 other than that minor changes implemented

**Comment by armstrmi** [comment]
@qingling128 @quentinmit everything should be updated just need to resolve the requested change from @quentinmit

File: confgenerator/testdata/valid/linux/metrics-receiver_activemq/input.yaml
Comment (qingling128) [inline confgenerator/testdata/valid/linux/metrics-receiver_activemq/input.yaml:26]: Looks like there is some merge conflict residue. Probably needs some cleanup.

File: confgenerator/testdata/valid/linux/metrics-receiver_activemq_no_jvm/input.yaml
Comment (qingling128) [inline confgenerator/testdata/valid/linux/metrics-receiver_activemq_no_jvm/input.yaml:27]: Same here

File: integration_test/third_party_apps_data/applications/activemq/centos_rhel/post
Comment (qingling128) [inline integration_test/third_party_apps_data/applications/activemq/centos_rhel/post:3]: Same here.

File: integration_test/third_party_apps_data/applications/activemq/debian_ubuntu/install
Comment (qingling128) [inline integration_test/third_party_apps_data/applications/activemq/debian_ubuntu/install:10]: Similar here.

File: integration_test/third_party_apps_data/applications/activemq/sles/post
Comment (qingling128) [inline integration_test/third_party_apps_data/applications/activemq/sles/post:3]: Another one.

**Review by qingling128 — COMMENTED** [review]
Some file diff shows merge conflict residues. I'm not sure why our unit tests did not catch them though. Weird.

File: integration_test/third_party_apps_data/applications/activemq/sles/post
Comment (armstrmi) [reply integration_test/third_party_apps_data/applications/activemq/sles/post]: Sorry about that, unsure how that happened as well, should be fixed now

Comment (qingling128) [reply integration_test/third_party_apps_data/applications/activemq/sles/post]: Thanks!

**Review by qingling128 — APPROVED** [review]
LGTM

3rd party tests all passed:
image

Needs rebasing before it can be merged.

Please rebase and let it automatically trigger unit tests. DO NOT need to re-trigger the integration tests. I'll merge it once the unit tests pass.

BTW in case there is some race condition between PRs and this one is out of sync again before I get back to it, could you grant maintainers access to rebase this PR by checking the [`Allow edits from maintainers`](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) button?

**Review by qingling128 — APPROVED** [review]
LGTM

3rd party tests all passed:
image

Needs rebasing before it can be merged.

Please rebase and let it automatically trigger unit tests. DO NOT need to re-trigger the integration tests. I'll merge it once the unit tests pass.

BTW in case there is some race condition between PRs and this one is out of sync again before I get back to it, could you grant maintainers access to rebase this PR by checking the [`Allow edits from maintainers`](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) button?

**Comment by armstrmi** [comment]
@qingling128 Perfect! all set!

**Pull request merged**
